HHS/Head-Start-TTADP
# [Prod] Improve training report validation visuals

## Description of change

- TTAHUB-1905:
A small PR, this moves some validation on the Training Report from from below the submit buttons to above the submit buttons (helping those with zoomed out screens not miss the alert). The new behavior is pictured below.

![Screenshot 2023-09-19 at 4 37 14 PM](https://github.com/HHS/Head-Start-TTADP/assets/3288586/811f588f-625e-49b0-92c1-f27b9e9d25a6)

This PR also improves the behavior when a user tries to "submit" a TR when they have a status other than Complete in the drop-down menu. Previously, an error would display but would not disappear when the status was changed to Complete (although submission would not be prevented with the correct status). Now, that is fixed.

- TTAHUB-1977: Use smartsheet ID for training report URLs
- TTAHUB-2118: Prevent changing root cause if on approved AR with conditions met.
- TTAHUB-1171: Add missing fields to review.
- TTAHUB-1860: Added performance logging to backend handlers. 
- TTAHUB-2100: Implemented AWS SES integration non-production environments.

## How to test

- TTAHUB-1977: Confirm that update, delete, view operations on training and session reports still work and the TR ID is derived from smartsheet
- TTAHUB-1905: Check that the above notes on Training Report validations are true.
- TTAHUB-2118: We should not be able to change the fei goal root cause if its on a approved report with at least one set.
- TTAHUB-1171: Missing fields should be on review.
- TTAHUB-1860: Performance logging is present.
- TTAHUB-2100: Emails to non-fake users works. Emails are not sent to fake email addresses.


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1905
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1171
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1860
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2100


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
